pre-commit-hooks/fix-local-includes.sh

```diff
@@ -2,26 +2,38 @@
 
 # Note: This script is intended to be run from the root of the repository.
 #
-# This script checks will fix local includes in the C++ code.
+# This script will fix local includes in the C++ code for a given file.
+# Usage: ./pre-commit-hooks/fix-local-includes.sh <file1> <file2> ...
 
-file_path="$1"
-
-echo "+ Fixing includes in $file_path..."
+files="$@"
+echo "+ Fixing includes in $files..."
 
 GNU_SED=$(sed --version 2>&1 | grep -q 'GNU' && echo true || echo false)
 
 if [[ "$GNU_SED" == "false" ]]; then # macOS sed
-    # make all includes to be <...> style
-    sed -i '' -E 's|#include "(.*)"|#include <\1>|g' "$file_path"
-
-    # make local includes to be "..." style
     main_src_dirs=$(find ./src -maxdepth 1 -type d -exec basename {} \; | tr '\n' '|' | sed 's/|$//' | sed 's/|/\\|/g')
-    sed -i '' -E "s|#include <(($main_src_dirs)/.*)>|#include \"\1\"|g" "$file_path"
 else
-    # make all includes to be <...> style
-    sed -i -E 's|#include "(.*)"|#include <\1>|g' "$file_path"
-
-    # make local includes to be "..." style
     main_src_dirs=$(find ./src -maxdepth 1 -type d  -exec basename {} \; | paste -sd '|' | sed 's/|/\\|/g')
-    sed -i -E "s|#include <(($main_src_dirs)/.*)>|#include \"\1\"|g" "$file_path"
 fi
+
+fix_includes() {
+    file_path="$1"
+
+    if [[ "$GNU_SED" == "false" ]]; then # macOS sed
+        # make all includes to be <...> style
+        sed -i '' -E 's|#include "(.*)"|#include <\1>|g' "$file_path"
+
+        # make local includes to be "..." style
+        sed -i '' -E "s|#include <(($main_src_dirs)/.*)>|#include \"\1\"|g" "$file_path"
+    else
+        # make all includes to be <...> style
+        sed -i -E 's|#include "(.*)"|#include <\1>|g' "$file_path"
+
+        # make local includes to be "..." style
+        sed -i -E "s|#include <(($main_src_dirs)/.*)>|#include \"\1\"|g" "$file_path"
+    fi
+}
+
+for file in $files; do
+    fix_includes "$file"
```

Comment:
Maybe this could be done in parallel now

Reply:
It's already done in parallel by `pre-commit` tool itself.
On my machine it spins up to 12 threads and runs this tool.
So, no need to think about executing in parallel.

And, since, we're only running on the changed files now because of `pre-commit` tool (when running locally), it is extremely fast now.